src/client/component/database.cpp

```diff
@@ -0,0 +1,60 @@
+#include <std_include.hpp>
+#include "loader/component_loader.hpp"
+
+#include "game/dvars.hpp"
+#include "game/game.hpp"
+
+#include <utils/io.hpp>
+#include <utils/hook.hpp>
+
+namespace database
+{
+	namespace
+	{
+		game::dvar_t* db_filesysImpl = nullptr;
+		utils::hook::detour db_fs_initialize_hook;
+
+		game::DB_FileSysInterface* db_fs_initialize_stub()
+		{
+			switch (db_filesysImpl->current.integer)
+			{
+			case 0:
+				return reinterpret_cast<game::DB_FileSysInterface*>(0x140BE82F8); // ptr to vtable of BnetTACTVFSManager (implements DB_FileSysInterface)
+			case 1:
+				return reinterpret_cast<game::DB_FileSysInterface*>(0x140BEFDC0); // ptr to vtable of DiskFS (implements DB_FileSysInterface)
+			default:
+				return nullptr; // this should not happen
+			}
+		}
+	}
+
+	class component final : public component_interface
+	{
+	public:
+		void post_unpack() override
+		{
+			static const char* values[] = 
+			{
+				"BnetTACTVFSManager", // (load files from CASC)
+				"DiskFS", // (load files from disk)
+				nullptr
+			};
+
+			const auto default_value = static_cast<int>(!utils::io::directory_exists("Data/data")
```

Comment:
This static cast is not necessary. Bool will be promoted to an int when call register enum.
https://docs.microsoft.com/en-us/cpp/cpp/type-conversions-and-type-safety-modern-cpp?view=msvc-170#widening-conversions-promotion
Just noticed this now 😅